projects/components/src/dropdown/dropdown-focus-handler.directive.ts

```diff
@@ -0,0 +1,165 @@
+ * ></vcd-dropdown>
+ */
+@Directive({
+    selector: 'vcd-dropdown[vcdDropdownFocusHandler]',
+})
+export class DropdownFocusHandlerDirective<T> implements AfterViewInit, OnDestroy {
+    constructor(
+        @Optional() @SkipSelf() private parentVcdDropdown: DropdownComponent<T>,
+        @Optional() @SkipSelf() private parentFocusHandler: DropdownFocusHandlerDirective<T>,
+        @Host() private hostVcdDropdown: DropdownComponent<T>,
+        private focusHandlerService: DropdownFocusHandlerService
+    ) {}
+
+    /**
+     * List of focusable menu items with their neighbors in 4 directions.
+     */
+    menuItems: MenuItem[];
+
+    private menuTrigger: MenuItem;
+    private isRootDropdown = !this.parentVcdDropdown;
+    private timeoutId: number;
+    private subscriptionTracker = new SubscriptionTracker(this);
+
+    /**
+     * After a dropdown menu is opened, it creates {@link MenuItem} for each of the menu items along with their trigger menu item and links
+     * them. It also then moves the focus to first item in the opened menu list.
+     */
+    ngAfterViewInit(): void {
+        this.subscriptionTracker.subscribe(this.hostVcdDropdown.dropdownMenuUpdated, (updated) => {
+            if (!updated) {
+                return;
+            }
+            this.createMenuTrigger();
+            this.linkMenuItems();
+            this.registerRootMenuContainer();
+            this.moveFocusToFirstItem();
+        });
+    }
+
+    ngOnDestroy(): void {}
+
+    private registerRootMenuContainer(): void {
+        if (this.isRootDropdown) {
+            const rootMenuContainer = this.hostVcdDropdown._clrDropdownMenuEl;
+            this.focusHandlerService.listenToArrowKeys(rootMenuContainer);
+        }
+    }
+
+    private moveFocusToFirstItem(): void {
+        this.focusHandlerService.moveFocusTo(this.menuTrigger);
+        // After the menu is opened, the ClrDropdownMenu wouldn't be attached to the DOM until the next change detection cycle.
+        // So setTimeout is the only way to wait for the ClrDropdownMenu to be ready to move focus to first item.
+        this.timeoutId = window.setTimeout(() => {
+            if (this.parentVcdDropdown) {
+                this.focusHandlerService.moveFocus(Direction.RIGHT);
+            } else {
+                this.focusHandlerService.moveFocus(Direction.DOWN);
+            }
+            window.clearTimeout(this.timeoutId);
+        });
+    }
+
+    private linkMenuItems(): void {
+        const menuChildren: Element[] = Array.from(this.hostVcdDropdown._clrDropdownMenuEl.children);
+        this.menuItems = menuChildren.map((child) => ({
+            element:
+                child.nodeName === BUTTON_NODE_NAME
+                    ? (child as HTMLElement)
+                    : (child.querySelector('clr-dropdown > button') as HTMLElement),
+        }));
+
+        this.linkVertical();
+
+        this.linkMenuToTrigger();
+    }
+
+    private createMenuTrigger(): void {
```

Comment:
Can we rename this to something like `findCurrentSelectedItem()`?

Reply:
I will add a comment to the menu trigger property